pkg/operator/ceph/file/mds/livenessprobe_test.go

```diff
@@ -0,0 +1,199 @@
+// acts as shim between go test definitions and bash code under test
+// define as string because embed doesn't work in _test.go files
+var probeShimWrapper string = `
+#!/usr/bin/env bash
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+export SCRIPT_DIR
+function ceph() {
+  # IMPORTANT: tests should always define behavior for specific commands being called to ensure code
+  # changes in the future do not inherit invalid success/failure behavior of existing unit tests
+  if [[ "$1" == "fs" ]] && [[ "$2" == "dump" ]] && [[ "$3" == "--mon-host=" ]] && [[ "$4" == "--mon-initial-members=" ]] && [[ "$5" == "--format" ]] && [[ "$6" == "json" ]]; then
+    [[ -n "$CEPH_MDS_DUMP_MOCK_STDOUT" ]] && echo "$CEPH_MDS_DUMP_MOCK_STDOUT"
+    [[ -n "$CEPH_MDS_DUMP_MOCK_STDERR" ]] && echo "$CEPH_MDS_DUMP_MOCK_STDERR" >/dev/stderr
+    exit $CEPH_MDS_DUMP_MOCK_RETCODE
+  fi
+  echo "command unexpected: ceph $*" >/dev/stderr
+  exit 111 # random return that probably won't overlap with anything else
+}
+export -f ceph
+function jq() {
+	$JQ "$@" # env should specify jq binary location
+}
+export -f jq
+#bash -x $SCRIPT_DIR/livenessprobe.sh
+`
+
+var (
+	//go:embed mdsProbeTestsFiles/livenessprobeTestOutput.json
+	testJsonData string
+)
+
+func Test_liveness_probe_script(t *testing.T) {
+	// liveness probe behavior has nuances that are critical to unit test, but there are challenges:
+	// - want to test the script in unit testing so tests live close to implementation
+	// - the probe script is written in bash, so shimming is required to test with go framework
+	// - the script processes output with 'jq'
+	//   -> effective tests will treat 'jq' as a black box to ensure 'jq' commands are correct
+	//   -> it is against good unit testing practice to require CLI tools for tests
+	// - a decent compromise is to skip this test in environments where 'jq' is not present (users)
+	//   BUT ensure the test always runs in CI
+
+	// TODO: set ROOK_UNIT_JQ_PATH in CI
+
+	jqPath := ""
+	jqPathOverride := os.Getenv("ROOK_UNIT_JQ_PATH") // set this env var to ensure this test runs in CI
+	if jqPathOverride != "" {
+		jqPath = jqPathOverride
+	} else {
+		// convenience behavior: allow unit test to run in user environments that have jq
+		cmd := exec.Command("which", "jq")
+		cmd.Env = os.Environ() // this is the ONLY circumstance where unit tests are allowed to use host's env
+		stdout := bytes.NewBuffer([]byte{})
+		cmd.Stdout = stdout
+		err := cmd.Run()
+		if err != nil {
+			jqPath = ""
+		} else {
+			jqPath = stdout.String()
+		}
+	}
+	if jqPath == "" {
+		// TODO: ensure this output doesn't show up in unit CI test output
+		t.Skip("skipping mds liveness probe script unit tests because jq binary location is not known")
+	}
+
+	t.Run("mds in map", func(t *testing.T) {
+		probe := generateMDSLivenessProbeExecDaemon("myfs-a", "myfs")
+		shellcmd := []string{"-c", probeShimWrapper}
+		shellcmd = append(shellcmd, probe.Exec.Command...)
```

Comment:
I am trying to run the probe command after that `probe.Exec.Command...`
But it is not executing,
@BlaineEXE any suggestions

The cmd is appending like this


```
...
export -f jq
#bash -x $SCRIPT_DIR/livenessprobe.sh
 timeout 20 sh -c #!/usr/bin/env bash
# do not use 'set -e -u' etc. because it is important to only fail this probe when failure is certain
# spurious failures risk destabilizing ceph or the filesystem

DAEMON_MDS="myfs-a"
FILESYSTEM_NAME="myfs"
...
```

Reply:
But after 
```
timeout 20 sh -c #!/usr/bin/env bash
# do not use 'set -e -u' etc. because it is important to only fail this probe when failure is certain
# spurious failures risk destabilizing ceph or the filesystem

DAEMON_MDS="myfs-a"
FILESYSTEM_NAME="myfs"
```
from which the probe cmd starts it is not executing, I tried to simply `exit 1` but it didn't ran

Reply:
I hope this helps answer: https://github.com/rook/rook/pull/12860/files#r1340187462